### ISSUE TITLE
Android app clears draft when switching windows

### ISSUE THREAD

**Issue by thenorili** [bluesky-social/social-app#3098]
**Describe the bug**

When switching windows on android with a draft open (eg switching to browser to get a URL) and returning to the bsky app, the draft is lost and the app switches to the home feed.

I previously assumed that this was the draft being evicted from memory (a solveable problem itself), however *critically* I can see the draft as it existed for a moment before the app refreshes and sends me to the home screen 

**To Reproduce**

1. Start a reply draft

2. Press the "change windows" button and switch to a browser. The state of the browser or the size of the webpage might matter?

3. Press the "change windows" button and return to bsky. You will briefly see the draft, but then the app will refresh and go to the home feed, losing the draft and the context of the reply (ie the post you were replying to)

**Expected behavior**

The draft's state should be maintained when changing windows with state dumped to disk if necessary so that the quotes or URLs can be added to it

**Screenshots**

**Details**

 - Platform: android app
 - Platform version: 14
 - App version: 1.70

**Additional context**

**Comment by aliceisjustplaying** [comment]
I tried reproducing it but couldn't. Did I do it right @thenorili ? Does it still happen on 1.71?

I tested 1.71 on Pixel 2 with Android 13 (LineageOS 20)


https://github.com/bluesky-social/social-app/assets/81575558/206a0c38-13f8-46c2-b7b3-e48e1149d1ad

**Comment by thenorili** [comment]
Yes, it still happens with the newest update. It isn't 100% -- if I quickly switch I can often get the page I left at, but if I take some time eg searching for a URL it resets to the home page.

I have a Galaxy A14 5G with Android 14.

**Comment by mozzius** [comment]
This is due to Android killing backgrounded apps to manage resources. I guess a proper fix for this would be to actually implement drafts 🙃

**Comment by mxmilkiib** [comment]
I've diligently tried to set Bluesky to not get killed, but sometimes it happens very soon after switching away from the app

it just killed a multi post composition for me

if what is being written automatically becomes a saved  draft then this would not happen

edit; if that's contentious, then make it a setting so people can avoid it. not everyone uses drafts the same way

**Comment by mozzius** [comment]
A fix for this is planned but no timeline at the moment
